File: api-aggregator/src/main/java/com/github/alexeysol/geekregime/apiaggregator/configurations/ModelMapperConfiguration.java
package com.github.alexeysol.geekregime.apiaggregator.configurations;

import org.modelmapper.Conditions;
import org.modelmapper.ModelMapper;
import org.modelmapper.convention.MatchingStrategies;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class ModelMapperConfiguration {
    @Bean
    public ModelMapper modelMapper() {
        ModelMapper modelMapper = new ModelMapper();

        modelMapper.getConfiguration()
            .setMatchingStrategy(MatchingStrategies.STRICT)
            .setSkipNullEnabled(true)
            .setPropertyCondition(Conditions.isNotNull());

        return modelMapper;
    }
}

File: src/main/kotlin/com/github/alexeysol/geekregimeapiusers/configurations/ModelMapperConfiguration.kt
package com.github.alexeysol.geekregimeapiusers.configurations

import org.modelmapper.Conditions
import org.modelmapper.ModelMapper
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class ModelMapperConfiguration {
    @Bean
    fun modelMapper(): ModelMapper {
        val modelMapper = ModelMapper()
        modelMapper.configuration.isSkipNullEnabled = true
        modelMapper.configuration.propertyCondition = Conditions.isNotNull()
        return modelMapper
    }
}
